src/components/mobile-nav-item/styles.ts

```diff
@@ -0,0 +1,96 @@
+import { styled } from '@linaria/react'
+
+export const ElMobileNavItemListItem = styled.li`
+  border-radius: inherit;
+`
+
+export const ElMobileNavSubItemUnorderedList = styled.ul`
+  &[aria-hidden='true'] {
+    display: none;
+  }
+`
+
+const baseStyles = `
+  cursor: pointer;
+  color: inherit;
+  display: flex;
+  border: none;
+  outline: none;
+  width: 100%;
+
+  border-radius: inherit;
+  border: 4px solid transparent;
+  background: inherit;
+  padding: var(--space-2, 8px) var(--space-4, 16px);
+
+  font-family: var(--font-family, Inter);
+  font-size: var(--font-size-base, 15px);
+  font-style: normal;
+  font-weight: var(--font-weight-regular, Regular);
+  line-height: var(--line-height-base, 24px);
+  letter-spacing: var(--letter-spacing-base, -0.15px);
+
+  &:hover {
+    background: var(--fill-default-light, #e5e9ed);
+  }
+
+  &:focus-visible {
+    border: 4px solid var(--purple-300, #7e9bfa);
+  }
+
+  &:active,
+  &[aria-current='true'],
+  &[aria-current='page'] {
+    color: var(--text-action, #4e56ea);
+  }
+`
+
+export const ElMobileNavItemAnchor = styled.a`
+  ${baseStyles}
+`
+
+export const ElMobileNavItemExpanderButton = styled.button`
+  ${baseStyles}
+`
+
+export const ElMobileNavItemContent = styled.span`
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  gap: var(--space-2, 8px);
+
+  flex-grow: 1;
+`
+
+export const ElMobileNavItemBadge = styled.span`
+  display: block;
+  width: var(--size-2, 8px);
+  height: var(--size-2, 8px);
+  background-color: var(--icon-error, #f01830);
+  border-radius: 100%;
+`
+
+export const ElMobileNavItemUnorderedList = styled.ul`
+  display: flex;
+  flex-direction: column;
+  background: var(--fill-white, #fff);
+  border-radius: var(--corner-lg, 8px);
+
+  &[data-is-expanded='true'] {
+    background: var(--fill-default-lightest, #f2f4f6);
+
+    ${ElMobileNavItemListItem} > ${ElMobileNavItemExpanderButton} {
+      border-radius: var(--corner-lg, 8px) var(--corner-lg, 8px) var(--corner-none, 0) var(--corner-none, 0);
+    }
+
+    ${ElMobileNavSubItemUnorderedList} > ${ElMobileNavItemListItem}:last-child {
+      border-radius: var(--corner-none, 0) var(--corner-none, 0) var(--corner-lg, 8px) var(--corner-lg, 8px);
+    }
```

Comment:
**suggestion:** you can avoid all this `border-radius` faff by placing the border-radius on the `<ul>` and setting `overflow: hidden` on the `<ul>` as well. This way, the `<li>` elements are effectively "clipped" by the `<ul>`s shape.

See https://codesandbox.io/p/sandbox/3jkrqc for what I mean. If I'm missing something with the CSS or DOM structure here, that's okay.

Reply:
> See https://codesandbox.io/p/sandbox/3jkrqc for what I mean

I tried to visit the link, but unfortunately I can't see anything on there; I'll try to implement about what you mean in the next commit 😉

Reply:
Sorry, it wasn't shared correctly. Should be accessible now.

Reply:
thanks, I got your point for that 👍

Reply:
Hi @kurtdoherty , thanks for the snippet. The purpose of defining or changing the border-radius is to modify the UI when it's expanded (applicable only to the expandable variant)

This would affect three sections:

1. Expandable trigger (top-left and top-right corners would have a corner)
2. Expandable content (no corners)
3. Expandable content's last item (bottom-left and bottom-right corners)

Please let me know if I'm misunderstanding your suggestion. It would be great if you could review the latest changes I've made, thanks

Reply:
Yes, I think you're misunderstanding the CodeSandbox I provided. Here, you're having to adjust the border radii of the elements within the three sections that you describe above.

My CodeSandbox is trying to demonstrate that you can take a simpler approach by defining the border radius of the parent element and hiding any overflow. This allows the child elements to have no border radius, but for the group as a whole to still have the correct radius visually (thanks to the parent).

If you still don't understand, it's not a big.

Reply:
ah yes, so the border radius is the only one that has the `border-radius` and `overflow` CSS properties, and then the rest of the child nodes would be rectangles. It's more simple agree

But with the current API, I think a refactor is required since the parent and the child nodes are actually the same component (`ElMobileNavItemListItem`). Perhaps using a data attribute could help us identify which is the parent and which is the child

Looking at the current condition, it seems we could leave it as it is for now and see how it works when design verifies the spec

Reply:
> But with the current API, I think a refactor is required since the parent and the child nodes are actually the same component (ElMobileNavItemListItem). Perhaps using a data attribute could help us identify which is the parent and which is the child

Yes, the data attribute is what distinguishes single items from expandable items, just like with the current CSS. All good though. This is all implementation detail that can be improved on in future if we need.